File: src/lib/alchemist/tests/basic1/basic1.ts
import { init, mut } from "sourcerer";

export class Basic_1 {
  myAccount: {
    data: u64;
  };

  @init("myAccount")
  initialize(data: u64) {
    this.myAccount.data = data;
  }

  @mut("myAccount")
  update(data: u64) {
    this.myAccount.data = data;
  }
}

File: src/lib/alchemist/tests/basic1/basic1.rs
use anchor_lang::prelude::*;

#[program]
mod basic_1 {
    use super::*;

    pub fn initialize(ctx: Context<Initialize>, data: u64) -> ProgramResult {
        ctx.accounts.my_account.data = data;
        Ok(())
    }

    pub fn update(ctx: Context<Update>, data: u64) -> ProgramResult {
        ctx.accounts.my_account.data = data;
        Ok(())
    }
}

#[derive(Accounts)]
pub struct Initialize<'info> {
    #[account(init)]
    pub my_account: ProgramAccount<'info, MyAccount>,
    pub rent: Sysvar<'info, Rent>,
}

#[derive(Accounts)]
pub struct Update<'info> {
    #[account(mut)]
    pub my_account: ProgramAccount<'info, MyAccount>,
}

#[account]
pub struct MyAccount {
    pub data: u64,
}
